runtime/near-vm-runner/src/wasmer_runner.rs

```diff
@@ -102,7 +100,9 @@ impl IntoVMError for wasmer_runtime::error::RuntimeError {
                 // invoke returns false and doesn't fill error info what Singlepass BE doesn't.
                 // Failed unwinder may happen in the case of deep recursion/stack overflow.
                 // Also can be thrown on unreachable instruction, which is quite unfortunate.
-                InvokeError::FailedWithNoError => VMError::FunctionCallError(WasmUnknownError),
+                InvokeError::FailedWithNoError => VMError::FunctionCallError(
+                    FunctionCallError::Nondeterministic("FailedWithNoError".to_string()),
+                ),
```

Comment:
Hm, the following two cases from the comments worry me: 

> Failed unwinder may happen in the case of deep recursion/stack overflow.
> Also can be thrown on unreachable instruction, which is quite unfortunate.

I see that we have tests for these cases, so presumably these don't result in non-deterministic errors, and the comment is wrong? Or is there some specific pattern of stack overflow which might cause us to always hit this case?

Reply:
It's called from two places: `call_func_with_index_inner` when `error_out` is None, or in `signal_trap_handler` when cannot unwind.